File: src/main/java/com/patres/timetable/repository/PeriodRepository.java
package com.patres.timetable.repository;

import com.patres.timetable.domain.Period;
import org.springframework.stereotype.Repository;

import org.springframework.data.jpa.repository.*;


/**
 * Spring Data JPA repository for the Period entity.
 */
@SuppressWarnings("unused")
@Repository
public interface PeriodRepository extends JpaRepository<Period,Long> {
    
}

File: src/main/kotlin/com/patres/timetable/repository/PeriodRepository.kt
package com.patres.timetable.repository

import com.patres.timetable.domain.Period
import com.patres.timetable.domain.Place
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface PeriodRepository : DivisionOwnerRepository<Period> {

    @Query("select period from Period period left join fetch period.intervalTimes where period.id =:id")
    fun findOneWithIntervals(@Param("id") id: Long?): Period?

}
